pac.c

```diff
@@ -91,54 +91,59 @@ static void fatal_handler(void *udata, const char *msg)
     logw("Fatal error: %s (%p).", msg, udata);
 }
 
+/*
+ * Note: error handling in these function is not standardized. The best
+ * documentation available is:
+ *
+ * https://msdn.microsoft.com/en-us/library/windows/desktop/gg308477(v=vs.85).aspx
+ *
+ * which states that (at least for the *Ex versions) the function should
+ * return an empty string if an error occurs (and not throw an error).
+ */
 static int dns_resolve(duk_context *ctx)
 {
     char buf[UTIL_BUFLEN];
     const char *host = duk_require_string(ctx, 0);
 
-    if (util_dns_resolve(host, buf, sizeof(buf), 0) < 0) {
-        return -1;
-    } else {
-        duk_push_string(ctx, buf);
-        return 1;
-    }
+    if (util_dns_resolve(host, buf, sizeof(buf), 0) < 0)
+        buf[0] = '\0';
+
+    duk_push_string(ctx, buf);
+    return 1;
```

Comment:
I'm good either way on this. In another language I'd say go for it, but when the only option for a flag (do the thing or don't do the thing) is an integer value it can be confusing to read something like this. But it looks like `util_dns_resolve` is doing something similar anyways. Maybe just use a more descriptive variable name than `i` and leave a short comment on what 0 or 1 does?